react/StoreLink.tsx

```diff
@@ -76,13 +76,8 @@ function StoreLink(props: Props) {
   const classes = useButtonClasses({ variant, size })
   const resolvedLink = useInterpolatedLink(href)
 
-  const [shouldReplaceUrl, setShouldReplaceUrl] = useState(
-    Boolean(modalDispatch)
-  )
-
-  useEffect(() => {
-    setShouldReplaceUrl(Boolean(modalDispatch))
-  }, [modalDispatch])
+  const hasModalContext = modalDispatch.toString() !== 'function () {}'
```

Comment:
How safe is that? I'm a bit concerned that this may come null or undefined somehow